envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/endpoints/EnvoyEndpointsFactory.kt

```diff
@@ -225,15 +225,12 @@ class EnvoyEndpointsFactory(
 
     private fun toEnvoyPriority(zone: String, locality: Locality): Int {
         val prioritiesProps = properties.loadBalancing.priorities
-        return when (prioritiesProps.zonePriorities.isNotEmpty()) {
-            true -> prioritiesProps.zonePriorities[zone] ?: toEnvoyPriority(locality)
-            false -> toEnvoyPriority(locality)
+        return if (locality == Locality.LOCAL) 0 else {
+            properties.loadBalancing.priorities.zonePriorities[zone] ?: 1
```

Comment:
So it would be like all to all configuration? 
Well as I understood, right now on-prem DCs should be always prioritised over non on-prem DCs, so in this case extended config that you are suggesting basically will always look the same as the one in the PR. 
I guess like from an open-source perspective it would be better making more configurable solution, but from what I see it requires kinda much more (maybe not much more but just more) changes, so I’m just not sure if it’s worth implementing it right now if it’s not required at the moment? Maybe when it’s the case then it would make sense to implement it?

Reply:
Yes, it will be an all-to-all configuration. If it doesn't require a lot of additional work, it would be beneficial to make it more configurable. The current configuration could potentially lead to misunderstandings. For example, according to this configuration, services deployed in the GCP zone should have a priority of 2. However, they will have a priority of 0 (for services deployed in GCP).

Reply:
done